indico/modules/events/registration/notifications.py

```diff
@@ -31,15 +31,16 @@ def notify_invitation(invitation, email_body, from_address):
     send_email(email, invitation.registration_form.event_new, 'Registration', session.user)
 
 
-def _notify_registration(registration, template, to_managers=False):
+def _notify_registration(registration, template, to_managers=False, from_address=None):
     template = get_template_module('events/registration/emails/{}'.format(template), registration=registration)
     to_list = registration.email if not to_managers else registration.registration_form.manager_notification_recipients
-    email = make_email(to_list=to_list, template=template, html=True)
+    email = make_email(to_list=to_list, template=template, html=True, from_address=from_address)
     send_email(email, event=registration.registration_form.event, module='Registration', user=session.user)
 
 
 def notify_registration_creation(registration):
-    _notify_registration(registration, 'registration_creation_to_registrant.html')
+    _notify_registration(registration, 'registration_creation_to_registrant.html',
+                         from_address=registration.registration_form.notification_sender_address)
```

Comment:
Why not do it inside `_notify_registration`? You need it for all the emails send to registrants, not just the creation email.